File: 0274-h-index/0274-h-index.java
class Solution {
    static int ans = 0;
    public int hIndex(int[] citations) { 
       int l = citations.length;
       Arrays.sort(citations);
       if (citations[l-1] == 0) return 0;
       bs(citations,0,l-1);
       return ans;
    }
    public void bs(int[] arr,int min,int max){
        if (min > max) return;
        int mid = (min+max)/2;
        if (arr[mid]>= arr.length - mid){
            ans = arr.length - mid;
            bs(arr,min,mid-1);
        } else {
            bs(arr,mid+1,max);
        }
    }
}

File: 0274-h-index/0274-h-index.ts
function hIndex(citations: number[]): number {
   let count = 0;
    for (let i = citations.length; i >= 1; i--){
        count = 0
        for (let j = 0; j < citations.length; j++){
            if (i <= citations[j]){
                count++
            }
        }
        if (count >= i){
            return i
        }
    }
    return 0
};